File: Week2/rkdtndk99/비밀지도.py
# 내 풀이
def solution(n, arr1, arr2):
    answer = []
    temp = ''
    for i in range(n):
        temp1 = format(arr1[i], "b").zfill(n)
        temp2 = format(arr2[i], "b").zfill(n)
        arr1[i] = temp1
        arr2[i] = temp2

    for j in range(n):
        for k in range(n):
            if arr1[j][k] == '1' or arr2[j][k] == '1':
                temp += '#'
            else:
                temp += ' '
        answer.append(temp)
        temp = ''

    return answer

# 다른 사람 풀이
def solution(n, arr1, arr2):
    answer = []
    for i, j in zip(arr1, arr2):
        temp = str(bin(i | j)[2:])
        temp = temp.rjust(n, '0')
        temp = temp.replace('1', '#')
        temp = temp.replace('0', ' ')
        answer.append(temp)

    return answer

File: Week2/alotofhee/비밀지도.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        for (int i = 0; i < n; i++) {
            answer[i] = Integer.toBinaryString(arr1[i] | arr2[i]); //2진수변환 후 OR연산 수행
            answer[i] = answer[i].replace('0', ' ');
            answer[i] = answer[i].replace('1', '#');
            while (answer[i].length() < n) { //공백 길이n까지 채우기 
                answer[i] = ' ' + answer[i];
            }
        }
        return answer;
    }
}
